devfiles/src/classes/widgets/BarChart.tsx

```diff
@@ -33,84 +33,33 @@ export default class BarChart extends TimeChart {
 
     // bind Chart specific options to Timechart Options
     public bindOptions(): void {
-        this.options = [this.xAxisSelector, this.yAxisSelector, ...this.colorOptions];
+        this.options = [this.xAxisSelector, this.yAxisSelector, this.colorOption];
     }
 
     public generateChartSpecificOptions(numTraces: number): void {
-        // Default Color palette
-        const palette = [
-            '#039cad',
-            '#ce4458',
-            '#d69c46',
-            '#5d7def',
-            '#70ce29',
-            '#fc7a2f',
-            '#2c48ba',
-            '#ce2b43',
-            '#ea4ba2',
-            '#5dbc29',
-            '#f26f52',
-            '#b7852d',
-            '#ea1932',
-            '#0ed80a',
-            '#46c60f',
-            '#dd5db2',
-            '#222e96',
-            '#ba0088',
-            '#fcf016',
-            '#930524',
-            '#d30684',
-            '#36d1ac',
-            '#f2dc15',
-            '#14706e',
-            '#092268',
-            '#1ab4ba',
-            '#aed33d',
-            '#adef51',
-            '#471687',
-            '#270799',
-            '#bc6ef4',
-            '#87c627',
-            '#f72ec1',
-            '#0345a3',
-            '#20137c',
-            '#e35ee5',
-            '#990f3f',
-            '#8af74f',
-            '#a441ea',
-            '#b210cc'
-        ];
-        // Change Number of Trace Options according to difference in length
-        const currentLen = this.colorOptions.length;
-        if (numTraces > this.colorOptions.length) {
-            for (let i = 0; i < numTraces - currentLen; i++) {
-                const colorOption = new ColorOption(
-                    this.panel,
-                    'color of trace ' + i.toString(),
-                    palette[(i + currentLen) % 40]
-                );
-                colorOption.extendedCallbacks.push(() => {
-                    this.refresh();
-                });
-                this.colorOptions.push(colorOption);
-            }
-        } else {
-            this.colorOptions = this.colorOptions.slice(0, numTraces);
+        if (this.colorOption == undefined) {
+            this.colorOption = new ColorOption(this.panel, 'color option', numTraces);
```

Comment:
Can the title be _Line/Bar/Plot Colours_ (up to you exactly which one)? _color option_ doesn't quite fit with the rest of the options

Reply:
To be changed in another commit.